onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -93,67 +128,39 @@ inline void CopyPrim(uint8_t* target, const uint8_t* source) {
   *reinterpret_cast<T*>(target) = *reinterpret_cast<const T*>(source);
 }
 
+// The function does not check num_axes > 0 but this is expected.
+template <class T>
+static void TypedDoTransposeEltWise(int64_t num_axes, const std::vector<int64_t>& target_dims, size_t num_blocks,
+                                    const std::vector<size_t>& stride, const uint8_t* source, uint8_t* target) {
+  std::vector<MultiIndex> mindex(num_axes);
+  size_t naxes = IncrementIndexAndComputeOffsetSetup(mindex.data(), num_axes, target_dims, stride, sizeof(T));
+
+  const uint8_t* local_source = source;
+  for (size_t i = 0; i < num_blocks; ++i) {
+    CopyPrim<T>(target, local_source);
+    IncrementIndexAndComputeOffset(mindex.data(), naxes, local_source);
```

Comment:
I think the most difficult part is how would you prove this code is correct? Here the local_source pointer may get moved randomly, how would you prove it won't be out of bound?

Reply:
I think the unit test did not catch this issue because it was compiled with pointer alignment but there should be a way to test that if one value is copied after an another one but at a lower position in the transposed array.

Reply:
I added a unit test to check that this function does not go outside the output vector.